### PR TITLE
Adding 3.6 compatibility + fixes

### PR DESCRIPTION
**Pull request by josbeir**
Ok so basically this a follow up from #155 but i decided to make the plugin fully 3.6 compatible. I'll be closing #155

Little changelog
* Fix resolving of document entities (this is broken on current master), the entityClass method was still doing a substr for the word 'Type' instead of 'Index', 4 vs 5 chars.
* Add registry aliases so DocumentContext works correctly, especially when using Indexes from plugins, these currently only work when setting the context type manually in FormHelper::create
* Cake 3.6 compatibility
  * Updated unit tests
  * added a few new get/set methods and added deprecation warnings on the old ones
  * Added a Plugin.php and removed the bootstrap.php file
* Add a Psr\Log\LoggerInterface compatible logger and pass data from ES to cake using this logger
* Default logging engine switched from Elastica\Log to cake's 'debug' engine. Falling back to Elastica\Log if not defined. Using Elastica\Log  in cake's configuration makes no sense right now because the log context is not used if the log message is a string.

### CODE REVIEW

**Comment by josbeir** [comment]
#152 should probably be closed in favor of this ? ;-)

File: README.md [hunk 1]
```diff
@@ -26,12 +26,19 @@ And run `php composer.phar update`
...
-Plugin::load('Cake/ElasticSearch', ['bootstrap' => true]);
+class Application extends BaseApplication
+{
+    public function bootstrap()
+    {
+        $this->addPlugin(ElasticSearchPlugin::class);
+
+        // If you want the plugin to automatically configure the Elastic model provider
+        // and FormHelper do the following:
+        // $this->addPlugin(ElasticSearchPlugin::class, [ 'bootstrap' => true ]);
```
Comment (markstory) [inline README.md]: Shouldn't the plugin class default bootstrap hook to on?

Comment (josbeir) [reply README.md]: I just adapted the current docs from current master, in my opinion it should indeed default to true.

File: tests/TestCase/Datasource/ConnectionTest.php [hunk 2]
```diff
@@ -88,4 +89,23 @@ public function testQueryLogging()
...
+        $logger = $this->getMockBuilder('Cake\Database\Log\QueryLogger')->setMethods(['log'])->getMock();
+        $logger->expects($this->once())->method('log');
+
+        $connection = ConnectionManager::get('test');
+        $connection->setLogger($logger);
+        $connection->logQueries(true);
+        $result = $connection->request('_stats');
+        $connection->logQueries(false);
+
+        $this->assertNotEmpty($result);
```
Comment (markstory) [inline tests/TestCase/Datasource/ConnectionTest.php]: Should there be assertions on the log message?

Comment (josbeir) [reply tests/TestCase/Datasource/ConnectionTest.php]: Done!

**Comment by lorenzo** [comment]
Thanks for a great job!

**Pull request merged**